pgmpy/factors/Factor.py

```diff
@@ -191,39 +190,22 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(), variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        marg_factor = []
+        for i in itertools.product(*[range(index) for index in assign[assign != -1]]):
+            assign[assign != -1] = i
+            temp_sum = np.sum(factor.values[factor._index_for_assignment(assign)])
+            marg_factor.append(temp_sum)
```

Comment:
@ankurankan please tell me the new-name for the `marg-factor` list so that I can make the appropriate changes.